openedx/core/djangoapps/password_policy/hibp.py

```diff
@@ -61,3 +68,18 @@ def range(password):
 
             except Exception as exc:  # pylint: disable=W0703
                 log.exception(f"Unable to range the password: {exc}")
+
+    @staticmethod
+    def is_sha1(maybe_sha):
+        """
+        Validates whether the provided string is sha1 encrypted or not
+        """
+        if len(maybe_sha) != SHA_LENGTH:
+            return False
+
+        try:
+            sha_int = int(maybe_sha, HEX_BASE)
+        except ValueError:
+            return True
```

Comment:
I guess we need to return False here if the provided string is not a valid sha1 hash.

Reply:
Good catch 👍